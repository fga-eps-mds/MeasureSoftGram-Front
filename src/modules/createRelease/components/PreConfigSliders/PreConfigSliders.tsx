/* eslint-disable no-unused-vars */
import React from 'react';
import { Grid, InputAdornment, Slider, SliderProps, TextField, TextFieldProps, Typography } from '@mui/material';
import capitalizer from '@utils/capitalizer';
import undelineRemover from '@utils/undelineRemover';
import PRE_CONFIG_SLIDERS from './consts';

interface PreConfigSlidersProps {
  label: string;
  onChange: Function;
  weight: number;
  disable: boolean;
}

const PreConfigSliders = ({ label, onChange, weight, disable }: PreConfigSlidersProps) => (
  <Grid spacing={3} container columns={16}>
    <Grid item xs={3}>
      <Typography textAlign="start">
        <strong>{capitalizer(undelineRemover(label))}</strong>
      </Typography>
    </Grid>
    <Grid item xs={9}>
      <Slider
        data-testid={PRE_CONFIG_SLIDERS.TEST_ID.CONFIG_SLIDER}
        value={weight}
        onChange={onChange as SliderProps['onChange']}
        disabled={disable}
        sx={{ marginLeft: '10px', marginRight: '10px' }}
      />
    </Grid>
    <Grid item xs={4}>
      <TextField
        data-testid={PRE_CONFIG_SLIDERS.TEST_ID.CONFIG_INPUT}
        sx={{ width: '96px', top: -4 }}
        variant="standard"
        value={weight}
        disabled={disable}
        onChange={onChange as TextFieldProps['onChange']}
        InputProps={{ endAdornment: <InputAdornment position="end">%</InputAdornment> }}
      />
    </Grid>
  </Grid>
);

export default PreConfigSliders;
