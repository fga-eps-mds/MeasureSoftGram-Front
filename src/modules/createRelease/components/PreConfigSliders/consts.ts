const PRE_CONFIG_SLIDERS = {
  TEST_ID: {
    CONFIG_SLIDER: 'CONFIG_SLIDER_TEST_ID',
    CONFIG_INPUT: 'CONFIG_INPUT_TEST_ID'
  }
};

export default PRE_CONFIG_SLIDERS;
