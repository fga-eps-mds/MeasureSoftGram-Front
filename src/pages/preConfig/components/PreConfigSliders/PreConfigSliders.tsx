/* eslint-disable no-unused-vars */
import React from 'react';
import { Grid, InputAdornment, Slider, SliderProps, TextField, TextFieldProps, Typography } from '@mui/material';
import capitilizer from '@utils/capitilizer';
import undelineRemover from '@utils/undelineRemover';
import PRE_CONFIG_SLIDERS from './consts';

interface PreConfigSlidersProps {
  label: string;
  onChange: Function;
  weight: number;
}

const PreConfigSliders = ({ label, onChange, weight }: PreConfigSlidersProps) => (
  <Grid spacing={3} container columns={16}>
    <Grid item xs={3}>
      <Typography textAlign="start">
        <strong>{capitilizer(undelineRemover(label))}</strong>
      </Typography>
    </Grid>
    <Grid item xs={9}>
      <Slider
        data-testid={PRE_CONFIG_SLIDERS.TEST_ID.CONFIG_SLIDER}
        value={weight}
        onChange={onChange as SliderProps['onChange']}
      />
    </Grid>
    <Grid item xs={4}>
      <TextField
        data-testid={PRE_CONFIG_SLIDERS.TEST_ID.CONFIG_INPUT}
        sx={{ width: '96px', top: -4 }}
        variant="standard"
        value={weight}
        onChange={onChange as TextFieldProps['onChange']}
        InputProps={{ endAdornment: <InputAdornment position="end">%</InputAdornment> }}
      />
    </Grid>
  </Grid>
);

export default PreConfigSliders;
