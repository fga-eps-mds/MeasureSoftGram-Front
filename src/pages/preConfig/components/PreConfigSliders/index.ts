import PreConfigSliders from './PreConfigSliders';

export default PreConfigSliders;
