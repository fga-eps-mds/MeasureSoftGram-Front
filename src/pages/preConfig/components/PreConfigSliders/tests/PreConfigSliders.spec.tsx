import '@testing-library/jest-dom';

import React from 'react';
import { render } from '@testing-library/react';

import PreConfigSliders from '../PreConfigSliders';

const LABEL = 'SLIDER';
const WEIGHT = 100;

describe('<PreConfigSliders />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.spyOn(console, 'error').mockImplementation(() => {});
    jest.spyOn(console, 'log').mockImplementation(() => {});
  });

  describe('Snapshot', () => {
    it('Deve corresponder ao Snapshot', () => {
      const tree = render(<PreConfigSliders label={LABEL} onChange={jest.fn()} weight={WEIGHT} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
