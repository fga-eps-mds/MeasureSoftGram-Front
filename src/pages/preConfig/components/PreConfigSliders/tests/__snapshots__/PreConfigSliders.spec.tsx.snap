// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreConfigSliders /> Snapshot Deve corresponder ao Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-12qstyp-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1v2d8t3-MuiTypography-root"
          >
            <strong>
              SLIDER
            </strong>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-trcag1-MuiGrid-root"
        >
          <span
            class="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root css-187mznn-MuiSlider-root"
          >
            <span
              class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
            />
            <span
              class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
              style="left: 0%; width: 100%;"
            />
            <span
              class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
              data-focusvisible="false"
              data-index="0"
              style="left: 100%;"
            >
              <input
                aria-orientation="horizontal"
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="100"
                data-index="0"
                max="100"
                min="0"
                step="1"
                style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                type="range"
                value="100"
              />
            </span>
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1equabv-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-rtrmyr-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1ptx2yq-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id=":r0:"
                type="text"
                value="100"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                >
                  %
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-12qstyp-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1v2d8t3-MuiTypography-root"
        >
          <strong>
            SLIDER
          </strong>
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-trcag1-MuiGrid-root"
      >
        <span
          class="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root css-187mznn-MuiSlider-root"
        >
          <span
            class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
          />
          <span
            class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
            style="left: 0%; width: 100%;"
          />
          <span
            class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
            data-focusvisible="false"
            data-index="0"
            style="left: 100%;"
          >
            <input
              aria-orientation="horizontal"
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="100"
              data-index="0"
              max="100"
              min="0"
              step="1"
              style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
              type="range"
              value="100"
            />
          </span>
        </span>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1equabv-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-rtrmyr-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1ptx2yq-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r0:"
              type="text"
              value="100"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
              >
                %
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
